shared/validate/validate.go

```diff
@@ -763,3 +763,46 @@ func IsAbsFilePath(value string) error {
 
 	return nil
 }
+
+func ParseNetworkVLANList(rawVLANValues []string) ([]string, error) {
```

Comment:
@zakufish please can you add a comment to this function explaining what the inputs and output formats are.

Reply:
If the output of this function is a slice of individual VLAN IDs, could it be made a `[]int` rather than `[]string` for clarity?

Reply:
The output of this particular function will be consumed as strings, for example:
```
for _, vlanID := range networkVLANList {
	// Reject VLAN ID 0 if specified (as validation allows VLAN ID 0 on unmanaged bridges for OVS).
	if vlanID == "0" {
		return fmt.Errorf("VLAN tagged ID 0 is not allowed for native Linux bridges")
```

Reply:
Another example (vlanIDs is the output of `ParseNetworkVLANList`):
`err = ovs.BridgePortSet(hostName, fmt.Sprintf("vlan_mode=%s", vlanMode), fmt.Sprintf("trunks=%s", strings.Join(vlanIDs, ",")))`

Reply:
> 

I see. But as the functions in the `validate` package do not concern themselves with the specific usage scenario, I would tend to prefer that the return values be used to better indicate the possible values being returned. Returning a `[]int` would make it very clear that the function is returning a list of individual VLAN IDs rather than anything else.

Then if the user of the functions wants them as a string, thats fine, they can be converted at that point by the caller of the function.